tests/suites/test_suite_x509write.function

```diff
@@ -39,25 +39,29 @@ static int x509_crt_verifycsr( const unsigned char *buf, size_t buflen )
     unsigned char hash[MBEDTLS_MD_MAX_SIZE];
     const mbedtls_md_info_t *md_info;
     mbedtls_x509_csr csr;
+    int ret;
+
+    mbedtls_x509_csr_init( &csr );
```

Comment:
It's not required to call init on csr before calling csr_parse() and friends, per our documentation, correct? Why add it here?

Reply:
It's required to call `mbedtls_x509_csr_free` in the cleanup code, which was added to fix a memory leak. Since `mbedtls_x509_csr_parse` is the very first thing that this function calls that might fail, the call to `mbedtls_x509_csr_init` is not necessary, but omitting it would be fragile: adding anything that can fail before `mbedtls_x509_csr_parse` would cause `mbedtls_x509_csr_free` to access uninitialized memory if that thing fails.